article/background.tex

```diff
@@ -4,107 +4,69 @@ \section{Background}
-However, recent advances in container technology have made this unnecessary, and it is now considered best practice to run containers without root privilages when reasonable.
-
-While the original reference OPFVTA article did not leverage this technology, containers can be used improve to improve the reliability and portability of the OPFVTA project.
-In this article, the authors will provide a snapshot of OPFVTA functioning at a certain point in time — mitigating process fragility in view of incrementing software dependency versions.
+Independent verification of published results is a crucial step in the establishment and maintanence of trust in shared scientific understanding. \cite{TODO Too corny to cite Kuhn?}.
+Verification research has become more common due in part to dramatic results \cite{TODO: Salmon}, dedicated communities \cite{TODO: something wholesome}, and widely publicized large-scale reproducibility meta-analyses \cite{RPP}.
+Unfortunately, in Neuroimaging, the execution of data analysis code often remains non-trivial and time consuming. \cite{TODO: I made this up} 
+The execution of data analysis often requires significant knowledge of their software dependencies, which should be considered out-of-scope for the expected requirements of a researcher to reexecute.
+Because the installation, configuration, and execution software environments frequently requires in-depth knowledge of the internal workings, reexecution results remain underrepresented in scientific literature \cite{TODO my opinion, is that ok?}
+
+\subsection{Complexity in Reexecution}
+
+Reexecution is an irreplacible component of research reproduction, but it frequently requires significant time and uncommon expertise to attempt.
+What makes reexecution particularly challenging is that it requires integration of many components, and even small differences in software environments can cause the entire system to fail.
+Therefore, the authors consider all code that is executed TODO:(not neccesarily OS,etc tone it down) to be part of the requisite analysis instructions.
+This includes code that is installed as dependencies, which may come from many repositories that cannot be relied upon to host all legacy versions.
+Reexecution either requires updating the software to modern dependencies, or the creation of a similar computational environment.
+Environment construction becomes harder to reproduce with time as upstream repositories update, as the experienced with this project.
+
+\subsection{Software Dependency Problem and Containers}
+
+A fundamental challenge of software dependency management is that dependencies can be nested arbitrarily deeply, which increases in complexity (TODO: O^n? does that even make sense to say?)
+As soon as one version of one dependency eventually becomes unavailable, the first step of executability is manually solving a computationally difficult problem.
```

Comment:
It's not a computational problem. If the source is gone you're boned, there's nothing computational or difficult.

Reply:
I'm specifically thinking of what happened when I had to bump one dependency and then I was trying to find a version that is compatible with the Python version, scipy, and numpy. Computationally difficult, I just mean that there arent really any great algorithms for solving a dependency problem like that other than brute force. +1 though, "computationally difficult" is not clear.